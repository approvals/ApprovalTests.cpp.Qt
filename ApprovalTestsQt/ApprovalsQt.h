#ifndef APPROVALTESTS_CPP_QT_APPROVALSQT_H
#define APPROVALTESTS_CPP_QT_APPROVALSQT_H

#include "ApprovalTests.hpp"
#include "ApprovalTestsQt/writers/QImageApprovalWriter.h"
#include "ApprovalTestsQt/writers/QTableWidgetWriter.h"

namespace ApprovalTestsQt
{
    void verifyQImage(const QImage& image,
        const ApprovalTests::Reporter& reporter = ApprovalTests::DiffReporter())
    {
        QImageApprovalWriter image_writer(image);
        ApprovalTests::Approvals::verify(image_writer, reporter);
    }

    void verifyQTableWidget(const QTableWidget& tableWidget,
        const ApprovalTests::Reporter& reporter = ApprovalTests::DiffReporter())
    {
        QTableWidgetWriter table_writer(tableWidget);
        ApprovalTests::Approvals::verify(table_writer, reporter);
    }
} // namespace ApprovalTestsQt

#endif //APPROVALTESTS_CPP_QT_APPROVALSQT_H
