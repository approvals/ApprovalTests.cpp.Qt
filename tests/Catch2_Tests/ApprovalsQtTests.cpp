#include "Catch.hpp"
#include <QImage>
#include <QTableWidget>
#include "ApprovalTestsQt/ApprovalsQt.h"

TEST_CASE("It approves a QImage")
{
    // begin-snippet: verify_qimage
    QImage image(10, 20, QImage::Format_RGB32);
    image.fill(Qt::red);
    ApprovalTestsQt::verifyQImage(image);
    // end-snippet
}

TEST_CASE("It approves a QTableWidget")
{
    QTableWidget tableWidget;
    const int rows = 2;
    const int columns = 3;
    tableWidget.setRowCount(rows);
    tableWidget.setColumnCount(columns);
    tableWidget.setHorizontalHeaderLabels(QStringList() << "Column 1"
                                                        << "Column 2"
                                                        << "Column 3");
    for (int i = 0; i != rows; ++i)
    {
        for (int j = 0; j != columns; ++j)
        {
            tableWidget.setItem(
                i, j, new QTableWidgetItem(QString::number((i + 1) * (j + 1))));
        }
    }
    ApprovalTestsQt::verifyQTableWidget(tableWidget);
}
