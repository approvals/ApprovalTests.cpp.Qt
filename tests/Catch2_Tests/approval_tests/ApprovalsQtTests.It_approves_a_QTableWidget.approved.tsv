"Column 1"	"Column 2"	"Column 3"
"1"	"2"	"3"
"2"	"4"	"6"
